doc/develop/test/pytest.rst

```diff
@@ -0,0 +1,84 @@
+.. integration-with-pytest:
+
+Integration with pytest test framework
+######################################
+
+*Please mind that integration of twister with pytest is still work in progress. Not every platform
```

Comment:
Can you please open a tacking issue for all things implemented and remaining items (preferably as a task list) to have an overview of what is done and what is coming?